docs/guides/latest/workloads/deployment.md

```diff
@@ -0,0 +1,435 @@
+      name: busybox
+    spec:
+      containers:
+      - args:
+        - sleep
+        - "3600"
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        name: busybox
+        volumeMounts:
+        - mountPath: /source/data
+          name: source-data
+      restartPolicy: Always
+      volumes:
+      - name: source-data
+        persistentVolumeClaim:
+          claimName: demo-pvc
+```
+
+Let's create the deployment we have shown above.
+
+```console
+$ kubectl apply -f ./recovered_deployment.yaml 
+persistentvolumeclaim/demo-pvc created
+deployment.apps/stash-recovered created
+```
+
+Now, wait for deployment’s pod to go into the `Running` state.
+
+```console
+$ kubectl get pod -n demo 
+NAME                               READY   STATUS    RESTARTS   AGE
+stash-recovered-676bd87957-7bcfg   1/1     Running   0          31s
+stash-recovered-676bd87957-pv4hg   1/1     Running   0          31s
+stash-recovered-676bd87957-q2hbl   1/1     Running   0          31s
+
+```
+
+**Create RestoreSession:**
+
+We need to create a `RestoreSession` crd to restore the backed up data in the target volume from backend.
+
+Below is the YAML of the `RestoreSesion` crd that we are going to create,
+
+```console
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: deployment-restore
+  namespace: demo
+spec:
+  repository:
+    name: local-repo
+  rules:
+    - paths:
+        - /source/data
+  target: # target indicates where the recovered data will be stored
+    ref:
+      apiVersion: apps/v1
+      kind: Deployment
+      name: stash-recovered
+    volumeMounts:
+      - name:  source-data
+        mountPath:  /source/data
+```
+
+Here,
+
+`spec.repository.name` specifies the `Repository` crd that holds the backend information where our backed up data has been stored.
+
+`spec.target.ref` refers to the target workload where the recovered data will be stored.
+
+Let's create the `RestoreSession` crd we have shown above,
+
+```console
+$ kubectl apply -f ./restoresession.yaml
+restoresession.stash.appscode.com/deployment-restore created
+```
+
+Once, you have created the `RestoreSession` crd, Stash will create a job to restore. We can watch the `RestoreSession` phase to check if the restore process is `succeeded` or not.
+
+Run the following command to watch RestoreSession phase,
+
+```console
+$ watch -n 2 kubectl get restore -n demo
+Every 5.0s: kubectl get restore -n demo           suaas-appscode: Mon Jun 24 10:33:57 2019
+
+NAME                 REPOSITORY-NAME   PHASE       AGE
+deployment-restore   local-repo        Succeeded   2m56s
+```
+
+So, we can see from the output of the above command that the restore process succeeded.
+
+**Verify Restored Data:**
+
+In this section, we will verify that the desired data has been restored successfully.
+
+Verify that the sample data has been restored in `/source/data` directory of the `stash-recovered` deployment's pod using the following command,
+
```

Comment:
First, find out the pod of `stash-recovered` deployment. Then run `exec` command.